llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -1757,6 +1794,35 @@ multiclass ATOM3P_impl<string AsmStr,  Intrinsic Intr,
 }
 
 // Constructs instrinsic name and instruction asm strings.
+multiclass ATOM_LdN_impl<string IntTypeStr, string TypeStr,
+                       string ScopeStr, string SpaceStr, string SemStr,
+                       NVPTXRegClass regclass, list<Predicate> Preds> {
+  defm : ATOM_LdP_impl<"ld." # SemStr
+                            # !if(!eq(SemStr, "volatile"), "", "." # ScopeStr)
+                            # !if(!eq(SpaceStr, "gen"), "", "." # SpaceStr)
+                            # "." # TypeStr
+                            # " \t$result, [$src];",
+                     !cast<Intrinsic>(
+                            "int_nvvm_ld_" # SpaceStr # "_" # IntTypeStr
+                            # !if(!eq(SemStr, "relaxed"), "", "_" # SemStr)
+                            # !if(!or(!eq(ScopeStr, "gpu"), !eq(ScopeStr, "")), "", "_" # ScopeStr)),
```

Comment:
Would it make sense to always explicitly use `gpu` suffix for the builtins/intrinsics?

Reply:
I think we should follow the pattern of existing builtins/intrinsics, which do not explicitly specify `gpu`.

Reply:
OK.